services/ux-backend/handlers/removecephclient/handler.go

```diff
@@ -0,0 +1,125 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, consumerName string) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl, consumerName)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client, consumerName string) {
+	klog.Info("POST method on /delete-ceph-client endpoint is invoked")
+	w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+	if message, err := removeCephZombieResources(ctx, cl, consumerName); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+		if _, err := w.Write([]byte("Failed to delete resources")); err != nil {
+			klog.Errorf("failed write data to response writer, %v", err)
+		}
+	} else {
+		klog.Info(message)
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+		if _, err = w.Write([]byte(message)); err != nil {
+			klog.Errorf("failed write data to response writer: %v", err)
+		}
+	}
+
+}
+
+func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
+	klog.Info("Only POST method should be used to send data to this endpoint /delete-ceph-client")
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+	w.Header().Set("Allow", "POST")
+
+	if _, err := w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method))); err != nil {
+		klog.Errorf("failed to write data to response writer: %v", err)
+	}
+}
+
+func removeCephZombieResources(ctx context.Context, cl client.Client, consumerName string) (string, error) {
+
+	storageconsumer := &ocsv1alpha1.StorageConsumer{}
+	err := cl.Get(context.Background(), types.NamespacedName{Name: consumerName,
+		Namespace: handlers.GetPodNamespace()}, storageconsumer)
+	if err != nil {
+		return "", fmt.Errorf("failed to get storage consumer with name: %s:%v", storageconsumer.Name, err)
+	}
+
+	storageclassrequests := &ocsv1alpha1.StorageClassRequestList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(handlers.GetPodNamespace()),
+		client.MatchingLabels{"ocs.openshift.io/storageconsumer-uuid": string(storageconsumer.UID)},
+	}
+	err = cl.List(ctx, storageclassrequests, listOpts...)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get list of storageclassrequests, %v", err)
+	}
+
+	if len(storageclassrequests.Items) == 0 {
+		return "", fmt.Errorf("no storageclassrequests present")
+	}
+	for _, scr := range storageclassrequests.Items {
+		for _, cephResourceSpec := range scr.Status.CephResources {
+			if cephResourceSpec.Kind == "CephBlockPoolRadosNamespace" {
+				radosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+				err = cl.Get(context.Background(), types.NamespacedName{Name: cephResourceSpec.Name,
```

Comment:
1. We can use the ctx passed in the function
2. Maybe we can use utils to construct the type? it just saves us from constructing a struct

Reply:
Okay